standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/properties/PropertyStore.java

```diff
@@ -0,0 +1,222 @@
+   * @param mapKey the map key
+   * @param getSchema the method to retrieve a schema if the map needs to be created
+   * @return the item property map
+   */
+  public abstract PropertyMap fetchProperties(String mapKey, Function<String, PropertySchema> getSchema);
+
+
+  /**
+   * Fetches a map of property maps.
+   * @param keyPrefix the map key prefix
+   * @param keyFilter a filter for map keys
+   * @param getSchema the method to retrieve a schema if the map needs to be created
+   * @return the map of property map
+   */
+  public abstract Map<String, PropertyMap> selectProperties(final String keyPrefix, Predicate<String> keyFilter, Function<String, PropertySchema> getSchema);
+  /**
+   * Fetches a property map digest.
+   * @param mapKey the map key
+   * @return the item property map
+   */
+  public abstract UUID fetchDigest(String mapKey);
+
+  /**
+   * Fetches a map of property maps digest keyed by their name.
+   * @param keyPrefix the map key prefix
+   * @param keyFilter a filter for map keys
+   * @return the map of property map digests
+   */
+  public abstract Map<String, UUID> selectDigest(String keyPrefix, Predicate<String> keyFilter);
+
+    /**
+     * Persists a property map.
+     * @param mapKey the map key
+     * @param map the map instance
+     */
+  protected abstract void saveProperties(String mapKey, PropertyMap map);
+
+
+  /**
+   * Drops a property map.
+   * @param mapKey the map key
+   */
+  protected abstract boolean dropProperties(String mapKey);
+
+  /**
+   * Renames a property map.
+   * @param mapKey the map source key
+   * @param newKey the new target key
+   */
+  public abstract boolean renameProperties(String mapKey, String newKey);
+
+  /**
+   * Persists an iterator property map.
+   * <p>May be useful to override to use one transaction.</p>
+   * @param save the iterator on pairs for map key, property map
+   */
+  public void saveProperties(Iterator<Map.Entry<String, PropertyMap>> save) {
+    while(save.hasNext()) {
+      Map.Entry<String, PropertyMap> pair = save.next();
+      PropertyMap map = pair.getValue();
+      if (map != null) {
+        saveProperties(pair.getKey(), map);
+      } else {
+        dropProperties(pair.getKey());
+      }
+    }
+  }
+
+  /**
+   * Serializes a map as a byte array.
+   * @param map the (nonnull) map to write
+   * @return the byte array
+   */
+  public byte[] serialize(PropertyMap map) {
+    return SerializationProxy.toBytes(map);
+  }
+
+  /**
+   * Deserializes a map from a byte array.
+   * @param bytes the byte array
+   * @return the (nonnull) oroperty map
+   */
+  public PropertyMap deserialize(byte[] bytes, Function<String, PropertySchema> getSchema) {
+    return SerializationProxy.fromBytes(bytes, this, getSchema);
+  }
+
+  /**
+   * Default ctor.
+   */
+  protected PropertyStore() {
+  }
+
+  /**
+   * A non-persistent store, for tests mainly.
+   */
+  public static class Transient extends PropertyStore {
```

Comment:
nit: If this is the class for test, could it move to the test package?

Reply:
It could. It most likely will. :-)

Reply:
Done.